### PR TITLE
Create PULL_REQUEST_TEMPLATE

### PR DESCRIPTION
**Pull request by oli-obk**
changelog: none

addresses https://github.com/rust-lang/rust-clippy/issues/3955#issuecomment-483567400

### CODE REVIEW

**Comment by flip1995** [comment]
While we're at it, we could maybe also add the [PR-checklist](https://github.com/rust-lang/rust-clippy/blob/master/doc/adding_lints.md#PR-Checklist) from the new-lint-documentation to the PR template with the note, that WIP PRs can be opened anytime.

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+We're collecting our changelog from pull request descriptions.
+If your PR only updates to the latest nightly, you can leave the
+`changelog` entry as `none`. Otherwise, please write a short comment
+explaining your change.
+
+If your PR fixes an issue, you can add "fixes #issue_number" into this
+PR description. This way the issue will be automatically closed when
+your PR is merged.
+
+Delete this line and everything above before opening your PR -->
```
Comment (mati865) [inline .github/PULL_REQUEST_TEMPLATE]: GitHub will ignore everything between `<!--` and `-->`, removing it is unnecessary

Comment (oli-obk) [reply .github/PULL_REQUEST_TEMPLATE]: Display-wise yes, but will it end up in the commit message?

**Comment by oli-obk** [comment]
> with the note, that WIP PRs can be opened anytime.

Users will only see that note if they are already opening a PR... not sure if the note would be helpful at that point.

File: .github/PULL_REQUEST_TEMPLATE (hunk 1 above)
Comment (mati865) [reply .github/PULL_REQUEST_TEMPLATE]: Right, forgot about bors commit messages.

**Comment by flip1995** [comment]
> Users will only see that note if they are already opening a PR

I see it as a reminder before pressing the "Open PR" button. Especially the `dev update_lints`, the documentation and `rustfmt` are things new contributors forget from time to time. Maybe we should add `rustfmt` to the new-lint-documentation checklist.

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] Executed `util/dev update_lints`
+- [ ] Added lint documentation
+- [ ] Run `cargo fmt`
+
+Note that you can skip the above if you are just opening a WIP PR in
+order to get feedback.
+
+Delete this line and everything above before opening your PR -->
+
+changelog: none
```
Comment (Manishearth) [inline .github/PULL_REQUEST_TEMPLATE]: Can this be in the actual PR body? Seems useful for it to be visible

Comment (oli-obk) [reply .github/PULL_REQUEST_TEMPLATE]: It is (outside the html-comment-block). Or am I misunderstanding what you're asking?

**Comment by Manishearth** [comment]
Never mind, reviewing on a phone is hard lol

**Pull request merged**